ml4h/explorations.py

```diff
@@ -691,6 +699,191 @@ def infer_with_pixels(args):
                 logging.info(f"Wrote:{stats['count']} rows of inference.  Last tensor:{tensor_paths[0]}")
 
 
+def _compute_masked_stats(img, y, nb_classes):
+    img = np.tile(img, nb_classes)
+    melt_shape = (img.shape[0], img.shape[1] * img.shape[2], img.shape[3])
+    img = img.reshape(melt_shape)
+    y = y.reshape(melt_shape)
+
+    masked_img = np.ma.array(img, mask=np.logical_not(y))
+    means = masked_img.mean(axis=1).data
+    medians = np.ma.median(masked_img, axis=1).data
+    stds = masked_img.std(axis=1).data
+    return means, medians, stds
+
+def _to_categorical(y, nb_classes):
+    return np.eye(nb_classes)[y]
+
+def _get_csv_row(sample_id, means, medians, stds, date):
+    res = np.concatenate([means, medians, stds], axis=-1)
+    csv_row = [sample_id] + res[0].astype('str').tolist() + [date]
+    return csv_row
+
+def _thresh_labels_above(y, img, intensity_thresh, in_labels, out_label, nb_orig_channels):
+    y = np.argmax(y, axis=-1)[..., np.newaxis]
+    y[np.logical_and(img >= intensity_thresh, np.isin(y, in_labels))] = out_label
+    y = y[..., 0]
+    y = _to_categorical(y, nb_orig_channels)
+    return y
+
+def infer_medians(args):
+    assert(args.batch_size == 1) # no support here for iterating over larger batches
+    assert(len(args.tensor_maps_in) == 1) # no support here for multiple input maps
+    assert(len(args.tensor_maps_out) == 1) # no support here for multiple output channels
+
+    tm_in = args.tensor_maps_in[0]
+    tm_out = args.tensor_maps_out[0]
+    assert(tm_in.shape[-1] == 1)  # no support here for stats on multiple input channels
+
+    # don't filter datasets for ground truth segmentations if we want to run inference on everything
+    # TODO HELP - this isn't giving me all 56K anymore
```

Comment:
Does this need to be addressed or can wait till re-ingest?

Reply:
It's up to us - the code doesn't currently work as it should right now. If args.analyze_ground_truth is True, then it should work on 56K tensors, but right now it still filters for only tensors that have ground truth. I had tried setting args.output_tensors and args.tensor_maps_out to empty lists in an attempt to override the code that filters for output tensors, but it's not currently working. That said, the project is in a waiting phase anyways now.